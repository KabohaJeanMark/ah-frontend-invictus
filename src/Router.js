import React, { Component } from 'react';
import { BrowserRouter, Route } from 'react-router-dom';
import { ToastContainer } from 'react-toastify';
import 'react-toastify/dist/ReactToastify.css';
import HomePage from '../src/app/components/Home';
import loginContainer from './app/containers/LoginContainer';
import RequestResetPassword from './app/containers/ResetPassword/RequestResetPassword';
import ResetPassword from './app/containers/ResetPassword/ResetPassword';
import SignupComponent from './app/containers/Signup';
import ArticleContainer from './app/containers/ArticleContainer';
import ProfileComponent from './app/containers/ProfileContainer';
import editProfileComponent from './app/containers/editProfileContainer';


class Router extends Component {
  render () {
    return (
      <BrowserRouter >
        <ToastContainer />
        <Route exact path="/" component={HomePage} />
        <Route path="/login" component={loginContainer} />
        <Route path="/password/reset" component={RequestResetPassword} />
        <Route path="/resetpassword/:token" component={ResetPassword} />
        <Route path="/signup" component={SignupComponent} />
        <Route path="/articles/create" component={ArticleContainer} />
        <Route path="/profile" component={ProfileComponent} />
        <Route path="/editprofile" component={editProfileComponent} />
      </BrowserRouter>
    );
  }
}

export default Router;
