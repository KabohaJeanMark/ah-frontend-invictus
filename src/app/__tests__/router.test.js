import React from 'react';
import toJson from 'enzyme-to-json';
import { shallow } from "enzyme";
import Router from '../../Router';

describe('<Router />', () => {
  let wrapper = shallow(<Router />);

  it('matches snapshot', () => {
    expect(toJson(wrapper)).toMatchSnapshot();
  });
});
