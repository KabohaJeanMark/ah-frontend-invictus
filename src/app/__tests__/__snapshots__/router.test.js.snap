// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Router /> matches snapshot 1`] = `
Array [
  <div
    class="Toastify"
  />,
  <div>
    <h5>
      Authors Haven
    </h5>
  </div>,
]
`;
